File: Anul II/Medii de proiectare si programare/Laborator/lab07/Ubb.BikeContest.Service/src/Service/TeamService.cs
using System.Collections;
using Ubb.BikeContest.Model;
using Ubb.BikeContest.Repository;
using Ubb.BikeContest.Services.validation;

namespace Ubb.BikeContest.Services.src.Service;

public class TeamService : ITeamService
{
    private readonly ITeamRepository _teamRepository;
    private readonly IValidator<long, Team> _validator;

    public TeamService(ITeamRepository teamRepository)
    {
        _teamRepository = teamRepository;
        _validator = new TeamValidator();
    }

    public Team FindById(long id)
    {
        return _teamRepository.FindById(id);
    }

    public IEnumerable<Team> FindAll()
    {
        return _teamRepository.FindAll();
    }

    public void Save(Team newEntity)
    {
        _validator.ValidateEntity(newEntity);
        _teamRepository.Save(newEntity);
    }

    public void Delete(long id)
    {
        _teamRepository.Delete(id);
    }

    public void Update(Team updatedEntity)
    {
        _validator.ValidateEntity(updatedEntity);
        _teamRepository.Update(updatedEntity);
    }

    public Team GetTeamByName(string teamName)
    {
        return _teamRepository.GetTeamByName(teamName);
    }
}

File: Anul II/Medii de proiectare si programare/Laborator/lab11/java/services/src/main/java/com/ubb/service/TeamService.java
package com.ubb.service;

import com.ubb.model.Team;
import com.ubb.repository.ITeamRepository;
import com.ubb.validation.IValidator;
import com.ubb.validation.TeamValidator;

import java.util.Collection;
import java.util.Optional;

public class TeamService implements ITeamService {
    private final ITeamRepository teamRepository;
    private final IValidator<Team> validator;

    public TeamService(final ITeamRepository teamRepository) {
        this.teamRepository = teamRepository;
        validator = new TeamValidator();
    }

    @Override
    public Optional<Team> findById(Long id) {
        return teamRepository.findById(id);
    }

    @Override
    public Collection<Team> findAll() {
        return teamRepository.findAll();
    }

    @Override
    public void save(Team newEntity) {
        validator.validateEntity(newEntity);
        teamRepository.save(newEntity);
    }

    @Override
    public void delete(Long id) {
        teamRepository.delete(id);
    }

    @Override
    public void update(Team updatedEntity) {
        teamRepository.update(updatedEntity);
    }

    @Override
    public Optional<Team> getTeamByName(String teamName) {
        return teamRepository.getTeamByName(teamName);
    }
}
